src/nnet3/nnet-diagnostics.h

```diff
@@ -100,6 +100,11 @@ class NnetComputeProb {
   // or NULL if there is no such info.
   const SimpleObjectiveInfo *GetObjective(const std::string &output_name) const;
 
+  // returns sum of objf and weights over all the SimpleObjectiveInfo objects,
+  // of what they computed.
+  // NnetCombiner::ComputeObjfAndDerivFromNnet() would use this function instead.
+  double GetTotalObjective(double *tot_weights) const;
```

Comment:
please rename tot_weights -> tot_weight here and in calling code.
And please be precise with language.  It returns the objf sum, and it outputs the
corresponding total weight to '*tot_weight'.  It doesn't return the total weight.